src/textUtils.c

```diff
@@ -171,30 +171,56 @@ size_t str_formatMetadata(const uint8_t* metadataHash, size_t metadataHashSize,
 	return encode_hex(metadataHash, metadataHashSize, out, outSize);
 }
 
-// check if it is ASCII between 32 and 126
-void str_validateTextBuffer(const uint8_t* text, size_t textSize)
+// check if a non-null-terminated buffer contains printable ASCII between 33 and 126 (inclusive)
+bool str_isPrintableAsciiWithoutSpaces(const uint8_t* buffer, size_t bufferSize)
 {
-	ASSERT(textSize < BUFFER_SIZE_PARANOIA);
+	ASSERT(bufferSize < BUFFER_SIZE_PARANOIA);
 
-	for (size_t i = 0; i < textSize; i++) {
-		VALIDATE(text[i] <= 126, ERR_INVALID_DATA);
-		VALIDATE(text[i] >= 32, ERR_INVALID_DATA);
+	for (size_t i = 0; i < bufferSize; i++) {
+		if (buffer[i] > 126) return false;
+		if (buffer[i] <  33) return false;
 	}
+
+	return true;
 }
 
-// check if a non-null-terminated buffer contains printable ASCII between 33 and 126 (inclusive)
-bool str_isAsciiPrintableBuffer(const uint8_t* buffer, size_t bufferSize)
+// check if a non-null-terminated buffer contains printable ASCII between 32 and 126 (inclusive)
+bool str_isPrintableAsciiWithSpaces(const uint8_t* buffer, size_t bufferSize)
 {
 	ASSERT(bufferSize < BUFFER_SIZE_PARANOIA);
 
 	for (size_t i = 0; i < bufferSize; i++) {
 		if (buffer[i] > 126) return false;
-		if (buffer[i] <  33) return false;
+		if (buffer[i] <  32) return false;
 	}
 
 	return true;
 }
 
+bool str_isAllowedDnsName(const uint8_t* buffer, size_t bufferSize)
+{
+	if (str_isPrintableAsciiWithSpaces(buffer, bufferSize)) {
```

Comment:
exiting early and returning false if there is some non-printable character would imho be a bit easier to follow